### ISSUE TITLE
Peer Evaluation Lab_02

### ISSUE THREAD

**Issue by NicoloIacobone** [figimodi/CI-2023#1]
The repository is well organised. The readme file is clear and it explains everything needed to understand the structure of the code.
The effort in research is huge: you fixed the original code and made it work properly (explaining why)!
I like the fact that you distinguished between Nim and Misere, they are the same game with just one rule of difference, but thanks to your explanation I understood that that rule makes an enormous difference in terms of strategy to play.
The evolutionary algorithm is implemented perfectly, I think that you really got the point, and you deeply understood the rules behind the behaviour of this family of algorithms. I love the way you use λ and τ, I couldn't imagine they could've been used.
I would have preferred some more comments in the code but the way the notebook's sections are divided is self-explanatory.
Illustrations are simple and effective, I suggest you to order them into a grid or to find a way to overlap them in order to have the entire knowledge of all the combinations you tested in a glance.
In short, you made an amazing job, it is clean, concise, and (it's not obvious) it works!
